File: app/src/main/kotlin/com/openclaw/ghostcrab/domain/model/OnboardingStep.kt
package com.openclaw.ghostcrab.domain.model

/**
 * Represents each step in the first-launch onboarding walkthrough.
 *
 * [Completed] is a terminal state that indicates onboarding is done;
 * it is excluded from the visible progress indicator count.
 */
public sealed interface OnboardingStep {
    public data object Welcome : OnboardingStep
    public data object WhatIsOpenClaw : OnboardingStep
    public data object InstallGateway : OnboardingStep
    public data object StartGateway : OnboardingStep
    public data object VerifyRunning : OnboardingStep
    public data object FindOnNetwork : OnboardingStep
    public data object Completed : OnboardingStep
}

/** Zero-based index of the step. [OnboardingStep.Completed] maps to 6. */
public val OnboardingStep.index: Int
    get() = when (this) {
        OnboardingStep.Welcome -> 0
        OnboardingStep.WhatIsOpenClaw -> 1
        OnboardingStep.InstallGateway -> 2
        OnboardingStep.StartGateway -> 3
        OnboardingStep.VerifyRunning -> 4
        OnboardingStep.FindOnNetwork -> 5
        OnboardingStep.Completed -> 6
    }

/** Total number of visible onboarding steps, excluding [OnboardingStep.Completed]. */
public const val ONBOARDING_STEPS_COUNT: Int = 6

File: ios/GhostCrab/Domain/Model/OnboardingStep.swift
import Foundation

/// Represents each step in the first-launch onboarding walkthrough.
///
/// `.completed` is a terminal state that indicates onboarding is done;
/// it is excluded from the visible progress indicator count.
public enum OnboardingStep: String, Codable, Sendable, CaseIterable, Hashable {
    case welcome
    case whatIsOpenClaw
    case installGateway
    case startGateway
    case verifyRunning
    case findOnNetwork
    case completed

    /// Zero-based index of the step. `.completed` maps to 6.
    public var index: Int {
        switch self {
        case .welcome: return 0
        case .whatIsOpenClaw: return 1
        case .installGateway: return 2
        case .startGateway: return 3
        case .verifyRunning: return 4
        case .findOnNetwork: return 5
        case .completed: return 6
        }
    }
}

/// Total number of visible onboarding steps, excluding `.completed`.
public let onboardingStepsCount: Int = 6
